sdk/storage/azure-storage-file-share/azure/storage/fileshare/_directory_client.py

```diff
@@ -591,9 +591,9 @@ def close_handle(self, handle: Union[str, "Handle"], **kwargs: Any) -> Dict[str,
             and the number of handles failed to close in a dict.
         :rtype: dict[str, int]
         """
-        try:
-            handle_id = handle.id  # type: ignore
-        except AttributeError:
+        if hasattr(handle, 'id'):
```

Comment:
Instead of this, use `isintance(handle, Handle)` as `Handle` should have an `id`. You will need to change the import of `Handle` to outside of `TYPE_CHECKING` and remove quotes from it above on line 577 and maybe elsewhere.